[CI] IpFilteringUpdateTests testThatInvalidDynamicIpFilterConfigurationIsRejected failing

Looks like these tests are trying to use an occupied port. We have various logic in our base test classes to ensure this doesn't happen but sounds like this test is doing something unique that's making this condition more likely. This happens across all tests in `IpFilteringUpdateTests` and I see similar errors in `SslMultiPortTests` as well.

**Build scan:**
https://gradle-enterprise.elastic.co/s/jiktat3ht3ye6/tests/:x-pack:plugin:security:internalClusterTest/org.elasticsearch.xpack.security.transport.filter.IpFilteringUpdateTests/testThatInvalidDynamicIpFilterConfigurationIsRejected


**Reproduction line:**
```
gradlew ':x-pack:plugin:security:internalClusterTest' --tests "org.elasticsearch.xpack.security.transport.filter.IpFilteringUpdateTests.testThatInvalidDynamicIpFilterConfigurationIsRejected" -Dtests.seed=DDEE06167C7FD912 -Dtests.locale=es-PE -Dtests.timezone=Etc/GMT+10 -Druntime.java=21
```

**Applicable branches:**
main, 8.11, 7.17

**Reproduces locally?:**
Didn't try

**Failure history:**
https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testThatInvalidDynamicIpFilterConfigurationIsRejected'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.security.transport.filter.IpFilteringUpdateTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium))))


**Failure excerpt:**
```
org.elasticsearch.transport.BindTransportException: Failed to bind to [::1]:[51805-51905]

  at __randomizedtesting.SeedInfo.seed([DDEE06167C7FD912:4F2E25CCAC535C51]:0)
  at org.elasticsearch.transport.TcpTransport.bindToPort(TcpTransport.java:453)
  at org.elasticsearch.transport.TcpTransport.bindServer(TcpTransport.java:414)
  at org.elasticsearch.transport.nio.NioTransport.doStart(NioTransport.java:101)
  at org.elasticsearch.xpack.security.transport.nio.SecurityNioTransport.doStart(SecurityNioTransport.java:113)
  at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:48)
  at org.elasticsearch.transport.TransportService.doStart(TransportService.java:318)
  at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:48)
  at org.elasticsearch.node.Node.start(Node.java:1176)
  at org.elasticsearch.test.InternalTestCluster$NodeAndClient.startNode(InternalTestCluster.java:1057)
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
  at java.util.concurrent.FutureTask.run(FutureTask.java:317)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:718)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  at java.lang.Thread.run(Thread.java:1583)

  Caused by: org.elasticsearch.common.util.concurrent.UncategorizedExecutionException: Failed execution

    at org.elasticsearch.common.util.concurrent.FutureUtils.rethrowExecutionException(FutureUtils.java:80)
    at org.elasticsearch.common.util.concurrent.FutureUtils.get(FutureUtils.java:50)
    at org.elasticsearch.action.support.AdapterActionFuture.actionGet(AdapterActionFuture.java:26)
    at org.elasticsearch.transport.nio.NioTransport.bind(NioTransport.java:76)
    at org.elasticsearch.transport.nio.NioTransport.bind(NioTransport.java:45)
    at org.elasticsearch.transport.TcpTransport.lambda$bindToPort$6(TcpTransport.java:441)
    at org.elasticsearch.common.transport.PortsRange.iterate(PortsRange.java:58)
    at org.elasticsearch.transport.TcpTransport.bindToPort(TcpTransport.java:439)
    at org.elasticsearch.transport.TcpTransport.bindServer(TcpTransport.java:414)
    at org.elasticsearch.transport.nio.NioTransport.doStart(NioTransport.java:101)
    at org.elasticsearch.xpack.security.transport.nio.SecurityNioTransport.doStart(SecurityNioTransport.java:113)
    at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:48)
    at org.elasticsearch.transport.TransportService.doStart(TransportService.java:318)
    at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:48)
    at org.elasticsearch.node.Node.start(Node.java:1176)
    at org.elasticsearch.test.InternalTestCluster$NodeAndClient.startNode(InternalTestCluster.java:1057)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
    at java.util.concurrent.FutureTask.run(FutureTask.java:317)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:718)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.lang.Thread.run(Thread.java:1583)

    Caused by: java.util.concurrent.ExecutionException: java.net.BindException: Failed to bind server socket channel {localAddress=/[0:0:0:0:0:0:0:1]:51905}.

      at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:257)
      at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:244)
      at org.elasticsearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:75)
      at org.elasticsearch.common.util.concurrent.FutureUtils.get(FutureUtils.java:45)
      at org.elasticsearch.action.support.AdapterActionFuture.actionGet(AdapterActionFuture.java:26)
      at org.elasticsearch.transport.nio.NioTransport.bind(NioTransport.java:76)
      at org.elasticsearch.transport.nio.NioTransport.bind(NioTransport.java:45)
      at org.elasticsearch.transport.TcpTransport.lambda$bindToPort$6(TcpTransport.java:441)
      at org.elasticsearch.common.transport.PortsRange.iterate(PortsRange.java:58)
      at org.elasticsearch.transport.TcpTransport.bindToPort(TcpTransport.java:439)
      at org.elasticsearch.transport.TcpTransport.bindServer(TcpTransport.java:414)
      at org.elasticsearch.transport.nio.NioTransport.doStart(NioTransport.java:101)
      at org.elasticsearch.xpack.security.transport.nio.SecurityNioTransport.doStart(SecurityNioTransport.java:113)
      at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:48)
      at org.elasticsearch.transport.TransportService.doStart(TransportService.java:318)
      at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:48)
      at org.elasticsearch.node.Node.start(Node.java:1176)
      at org.elasticsearch.test.InternalTestCluster$NodeAndClient.startNode(InternalTestCluster.java:1057)
      at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
      at java.util.concurrent.FutureTask.run(FutureTask.java:317)
      at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:718)
      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
      at java.lang.Thread.run(Thread.java:1583)

      Caused by: java.net.BindException: Failed to bind server socket channel {localAddress=/[0:0:0:0:0:0:0:1]:51905}.

        at org.elasticsearch.nio.ServerChannelContext.register(ServerChannelContext.java:76)
        at org.elasticsearch.nio.EventHandler.handleRegistration(EventHandler.java:54)
        at org.elasticsearch.nio.NioSelector.registerChannel(NioSelector.java:443)
        at org.elasticsearch.nio.NioSelector.setUpNewChannels(NioSelector.java:435)
        at org.elasticsearch.nio.NioSelector.preSelect(NioSelector.java:256)
        at org.elasticsearch.nio.NioSelector.singleLoop(NioSelector.java:149)
        at org.elasticsearch.nio.NioSelector.runLoop(NioSelector.java:125)
        at java.lang.Thread.run(Thread.java:1583)

        Caused by: java.net.BindException: Address already in use: bind

          at sun.nio.ch.Net.bind0(Net.java:-2)
          at sun.nio.ch.Net.bind(Net.java:565)
          at sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:344)
          at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:301)
          at java.nio.channels.ServerSocketChannel.bind(ServerSocketChannel.java:224)
          at org.elasticsearch.nio.ServerChannelContext.register(ServerChannelContext.java:73)
          at org.elasticsearch.nio.EventHandler.handleRegistration(EventHandler.java:54)
          at org.elasticsearch.nio.NioSelector.registerChannel(NioSelector.java:443)
          at org.elasticsearch.nio.NioSelector.setUpNewChannels(NioSelector.java:435)
          at org.elasticsearch.nio.NioSelector.preSelect(NioSelector.java:256)
          at org.elasticsearch.nio.NioSelector.singleLoop(NioSelector.java:149)
          at org.elasticsearch.nio.NioSelector.runLoop(NioSelector.java:125)
          at java.lang.Thread.run(Thread.java:1583)

```

--- a/modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/transport/netty4/Netty4TransportMultiPortIntegrationIT.java
+++ b/modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/transport/netty4/Netty4TransportMultiPortIntegrationIT.java
@@ -7,6 +7,7 @@
  */
 package org.elasticsearch.transport.netty4;
 
+import org.apache.lucene.util.Constants;
 import org.elasticsearch.ESNetty4IntegTestCase;
 import org.elasticsearch.action.admin.cluster.node.info.NodeInfo;
 import org.elasticsearch.action.admin.cluster.node.info.NodesInfoResponse;
@@ -31,14 +32,16 @@ import static org.hamcrest.Matchers.lessThanOrEqualTo;
 @ClusterScope(scope = Scope.SUITE, supportsDedicatedMasters = false, numDataNodes = 1, numClientNodes = 0)
 public class Netty4TransportMultiPortIntegrationIT extends ESNetty4IntegTestCase {
 
+    private static final int NUMBER_OF_CLIENT_PORTS = Constants.WINDOWS ? 300 : 10;
+
     private static int randomPort = -1;
     private static String randomPortRange;
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
         if (randomPort == -1) {
-            randomPort = randomIntBetween(49152, 65525);
-            randomPortRange = Strings.format("%s-%s", randomPort, randomPort + 10);
+            randomPort = randomIntBetween(49152, 65535 - NUMBER_OF_CLIENT_PORTS);
+            randomPortRange = Strings.format("%s-%s", randomPort, randomPort + NUMBER_OF_CLIENT_PORTS);
         }
         Settings.Builder builder = Settings.builder()
             .put(super.nodeSettings(nodeOrdinal, otherSettings))

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/pki/PkiOptionalClientAuthTests.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/pki/PkiOptionalClientAuthTests.java
@@ -41,7 +41,7 @@ public class PkiOptionalClientAuthTests extends SecuritySingleNodeTestCase {
 
     @BeforeClass
     public static void initPort() {
-        randomClientPort = randomIntBetween(49000, 65535 - NUMBER_OF_CLIENT_PORTS);
+        randomClientPort = randomIntBetween(49152, 65535 - NUMBER_OF_CLIENT_PORTS);
     }
 
     @Override

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/pki/PkiOptionalClientAuthTests.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/pki/PkiOptionalClientAuthTests.java
@@ -7,6 +7,7 @@
 package org.elasticsearch.xpack.security.authc.pki;
 
 import org.apache.http.nio.conn.ssl.SSLIOSessionStrategy;
+import org.apache.lucene.util.Constants;
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.RequestOptions;
 import org.elasticsearch.client.Response;
@@ -34,11 +35,13 @@ import static org.hamcrest.Matchers.is;
 
 public class PkiOptionalClientAuthTests extends SecuritySingleNodeTestCase {
 
+    private static final int NUMBER_OF_CLIENT_PORTS = Constants.WINDOWS ? 300 : 100;
+
     private static int randomClientPort;
 
     @BeforeClass
     public static void initPort() {
-        randomClientPort = randomIntBetween(49000, 65500);
+        randomClientPort = randomIntBetween(49000, 65535 - NUMBER_OF_CLIENT_PORTS);
     }
 
     @Override
@@ -47,7 +50,7 @@ public class PkiOptionalClientAuthTests extends SecuritySingleNodeTestCase {
     }
 
     protected Settings nodeSettings() {
-        String randomClientPortRange = randomClientPort + "-" + (randomClientPort + 100);
+        String randomClientPortRange = randomClientPort + "-" + (randomClientPort + NUMBER_OF_CLIENT_PORTS);
 
         Settings.Builder builder = Settings.builder()
             .put(super.nodeSettings())

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/transport/filter/IpFilteringIntegrationTests.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/transport/filter/IpFilteringIntegrationTests.java
@@ -6,6 +6,7 @@
  */
 package org.elasticsearch.xpack.security.transport.filter;
 
+import org.apache.lucene.util.Constants;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.transport.TransportAddress;
 import org.elasticsearch.core.SuppressForbidden;
@@ -28,11 +29,14 @@ import static org.hamcrest.Matchers.is;
 // no client nodes as they all get rejected on network connections
 @ClusterScope(scope = Scope.SUITE, numDataNodes = 0, numClientNodes = 0)
 public class IpFilteringIntegrationTests extends SecurityIntegTestCase {
+
+    private static final int NUMBER_OF_CLIENT_PORTS = Constants.WINDOWS ? 300 : 100;
+
     private static int randomClientPort;
 
     @BeforeClass
     public static void getRandomPort() {
-        randomClientPort = randomIntBetween(49000, 65500); // ephemeral port
+        randomClientPort = randomIntBetween(49152, 65535 - NUMBER_OF_CLIENT_PORTS); // ephemeral port
     }
 
     @Override
@@ -42,7 +46,7 @@ public class IpFilteringIntegrationTests extends SecurityIntegTestCase {
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
-        String randomClientPortRange = randomClientPort + "-" + (randomClientPort + 100);
+        String randomClientPortRange = randomClientPort + "-" + (randomClientPort + NUMBER_OF_CLIENT_PORTS);
         return Settings.builder()
             .put(super.nodeSettings(nodeOrdinal, otherSettings))
             .put("transport.profiles.client.port", randomClientPortRange)

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/transport/filter/IpFilteringUpdateTests.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/transport/filter/IpFilteringUpdateTests.java
@@ -6,6 +6,7 @@
  */
 package org.elasticsearch.xpack.security.transport.filter;
 
+import org.apache.lucene.util.Constants;
 import org.elasticsearch.cluster.ClusterState;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.Strings;
@@ -26,13 +27,15 @@ import static org.hamcrest.Matchers.is;
 @ClusterScope(scope = TEST, supportsDedicatedMasters = false, numDataNodes = 1)
 public class IpFilteringUpdateTests extends SecurityIntegTestCase {
 
+    private static final int NUMBER_OF_CLIENT_PORTS = Constants.WINDOWS ? 300 : 100;
+
     private static int randomClientPort;
 
     private final boolean httpEnabled = randomBoolean();
 
     @BeforeClass
     public static void getRandomPort() {
-        randomClientPort = randomIntBetween(49000, 65500);
+        randomClientPort = randomIntBetween(49152, 65535 - NUMBER_OF_CLIENT_PORTS);
     }
 
     @Override
@@ -42,7 +45,7 @@ public class IpFilteringUpdateTests extends SecurityIntegTestCase {
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
-        String randomClientPortRange = randomClientPort + "-" + (randomClientPort + 100);
+        String randomClientPortRange = randomClientPort + "-" + (randomClientPort + NUMBER_OF_CLIENT_PORTS);
         return Settings.builder()
             .put(super.nodeSettings(nodeOrdinal, otherSettings))
             .put("xpack.security.transport.filter.deny", "127.0.0.200")

